src/Database/Redis/Cluster.hs

```diff
@@ -172,11 +186,26 @@ evaluatePipeline :: MVar ShardMap -> IO ShardMap -> Connection -> [[B.ByteString
 evaluatePipeline shardMapVar refreshShardmapAction conn requests = do
         shardMap <- hasLocked "reading shardmap in evaluatePipeline" $ readMVar shardMapVar
         requestsByNode <- getRequestsByNode shardMap
-        resps <- concat <$> mapM (uncurry executeRequests) requestsByNode
+        -- catch the exception thrown at each node level
+        -- send the command to random node.
+        -- merge the current responses with new responses.
+        eresps <- mapM (try . uncurry executeRequests) requestsByNode
+        -- take a random connection where there are no exceptions.
+        let (nc, _) = (requestsByNode !!) $ fromMaybe 0 $ findIndex isRight eresps
```

Comment:
Need to figure out a way to chose the random node than `0`. Still this should be okay because it would fail the current queries in pipeline.